File: peel-core/src/main/java/eu/stratosphere/peel/core/beans/StringToLifespan.java
package eu.stratosphere.peel.core.beans;

import eu.stratosphere.peel.core.beans.system.Lifespan;
import org.springframework.core.convert.converter.Converter;
import scala.Enumeration.Value;

public class StringToLifespan implements Converter<String, Value> {

    @Override
    public Value convert(String s) {
        if (s.equals("PROVIDED")) {
            return Lifespan.PROVIDED();
        } else if (s.equals("SUITE")) {
            return Lifespan.SUITE();
        } else if (s.equals("EXPERIMENT")) {
            return Lifespan.EXPERIMENT();
        } else {
            throw new IllegalArgumentException(s + " can not be converted to Scala Lifecycle Value!");
        }
    }
}

File: peel-core/src/main/scala/eu/stratosphere/peel/core/beans/StringToLifespan.scala
package eu.stratosphere.peel.core.beans

import eu.stratosphere.peel.core.beans.system.Lifespan
import org.springframework.core.convert.converter.Converter

/** Spring Converter to convert Java Strings to [[eu.stratosphere.peel.core.beans.system.Lifespan Lifespan]] Values
  *
  */
class StringToLifespan extends Converter[String, Lifespan.Value]{

  def convert(s: String): Lifespan.Value = {
    if (s == "PROVIDED") {
      Lifespan.PROVIDED
    }
    else if (s == "SUITE") {
      Lifespan.SUITE
    }
    else if (s == "EXPERIMENT") {
      Lifespan.EXPERIMENT
    }
    else if (s == "JOB") {
      Lifespan.JOB
    }
    else {
      throw new IllegalArgumentException(s + " can not be converted to Scala Lifecycle Value!")
    }
  }
}
